### PR TITLE
Add exclude parameter to replication script

### PR DESCRIPTION
**Pull request by cbickel**
This PR adds the ability to exclude some databases from the backup with the exclude parameter.

In addition it improves deletion of old replication design documents.

### CODE REVIEW

**Comment by markusthoemmes** [comment]
Description please of what is implemented and why.

File: tools/db/replicateDbs.py [hunk 1]
```diff
@@ -72,7 +77,9 @@ def isExpired(timestamp):
     print("----- Delete backup-documents older than %d seconds -----" % args.expires)
     for doc in filter(lambda doc: isBackupDb(doc.id) and isExpired(extractTimestamp(doc.id)), replicator.view('_all_docs', include_docs=True)):
         print("deleting backup document: %s" % doc.id)
-        replicator.delete(doc.doc)
+        # Get again the latest version of the document to delete the right revision and avoid Conflicts
+        docToDelete = replicator[doc.id]
+        replicator.delete(docToDelete)
```
Comment (sven-lange-last) [inline tools/db/replicateDbs.py]: There should be a retry around fetch / delete to make it more robust.

**Comment by cbickel** [comment]
PG1#1473 is green.

File: tools/db/replicateDbs.py [hunk 2]
```diff
@@ -25,12 +25,24 @@
...
 
+def retry(fn, retries, *args):
+    try:
+        return fn(*args)
+    except:
+        if (retries > 0):
+            time.sleep(1)
+            return retry(fn, retries - 1, *args)
+        else:
+            raise
```
Comment (markusthoemmes) [inline tools/db/replicateDbs.py]: As noted in chat, `retry` should take an applied function (`Function0`) achieveable for example using `lambda`.

File: tools/db/replicateDbs.py [hunk 3]
```diff
@@ -108,6 +124,7 @@ def replayDatabases(args):
 replicateParser.add_argument("--dbPrefix", required=True, help="Prefix of the databases, that should be backed up.")
 replicateParser.add_argument("--expires", required=True, type=int, help="Deletes all backups, that are older than the given value in seconds.")
 replicateParser.add_argument("--continuous", action="store_true", help="Wether or not the backup should be continuous")
+replicateParser.add_argument("--exclude", default="", help="Comma separared list of database names, that should not be backed up. (Without prefix).")
```
Comment (markusthoemmes) [inline tools/db/replicateDbs.py]: comma separated

**Pull request merged**